controller/tap/server.go

```diff
@@ -30,23 +29,24 @@ const requireIDHeader = "l5d-require-id"
 const podIPIndex = "ip"
 const defaultMaxRps = 100.0
 
-type (
-	server struct {
-		tapPort             uint
-		k8sAPI              *k8s.API
-		controllerNamespace string
-	}
-)
+// GRPCTapServer describes the gRPC server implementing tap.Tap_TapByResourceServer
```

Comment:
It's worth considering if we want to continue having this implement the TapByResource gRPC interface.  Since we only call these methods directly, it's not strictly necessary to continue to implement that interface.  Continuing to implement the interface would make it easier in the future if we ever were to split this back out into a gRPC server again.  On the other hand, breaking that interface would allow us to greatly simplify this code and remove the need for the ServerStream boilerplate.

My preference would be to break the interface and simplify the code, but it's worth considering the tradeoffs.

Reply:
@olix0r do you have thoughts about maintaining this interface?

Reply:
@adleong I'm sorry, there are like 3 interfaces at play here. Which interface are we talking about changing? What uses it?

Reply:
GRPCTapServer implements the TapServer interface which is autogenerated from the (deprecated) protobuf service definition.  Since this is not used as a gRPC server, nothing uses this interface.  

My suggestion is that we change the signature of TapByResource to be more ergonomic (for example by taking a channel instead of a `pb.Tap_TapByResourceServer`) but this would mean that we no longer implement the TapServer interface.

Reply:
In order to keep the existing gRPC interface for use in the future, we're going to keep these around.